pymc3/sampling.py

```diff
@@ -1070,6 +1083,12 @@ def sample_posterior_predictive(trace, samples=None, model=None, vars=None, size
 
     model = modelcontext(model)
 
+    if varnames is not None:
+        if vars is not None:
+            raise Exception("Should not specify both vars and varnames arguments.")
+        vars = (model[x] for x in varnames)
+    else:
+        raise DeprecationWarning("vars argument is deprecated in favor of varnames.")
```

Comment:
Use the [warnings package](https://docs.python.org/3.6/library/warnings.html) instead of raising the `DeprecationWarning`. Also, this warning should be raised if `varnames is None` and `vars is not None`.

Reply:
Got this now, I believe.  Waiting for the tests to complete.
There's enough uninteresting history on this topic branch that it's probably worth squash-merging it, or I could clean up the history.  Need to rebase it, anyway.